app/forms/steps/client/appeal_reference_number_form.rb

```diff
@@ -0,0 +1,44 @@
+module Steps
+  module Client
+    class AppealReferenceNumberForm < Steps::BaseFormObject
+      include Steps::HasOneAssociation
+      has_one_association :case
+
+      MAAT_ID_REGEXP = /\A[0-9]{6,9}\z/
+
+      attribute :appeal_reference_number, :string
+      attribute :appeal_maat_id, :string
+      attribute :appeal_usn, :string
+
+      validates :appeal_reference_number, presence: true
+      validates :appeal_maat_id, presence: true, if: :maat_id_selected?
+      validates :appeal_usn, presence: true, if: :usn_selected?
+
+      validates :appeal_maat_id,
+                format: { with: MAAT_ID_REGEXP }, allow_blank: true
+
+      private
+
+      def maat_id_selected?
+        appeal_reference_number == 'appeal_maat_id'
+      end
+
+      def usn_selected?
+        appeal_reference_number == 'appeal_usn'
+      end
+
+      def persist!
+        self.case.update(
+          attributes.merge(attributes_to_reset)
+        )
+      end
+
+      def attributes_to_reset
+        {
+          'appeal_maat_id' => (appeal_maat_id unless usn_selected?),
+          'appeal_usn' => (appeal_usn unless maat_id_selected?)
+        }
+      end
```

Comment:
I think instead of this you could take the approach of the [premium bonds](https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/blob/main/app/forms/steps/capital/premium_bonds_form.rb#L26) form which would reset the maat id or the usn if the other option is selected before saving and submitting and then i don't think you would need this method. Let me know if that makes sense